net/dns/manager_windows.go

```diff
@@ -134,53 +143,35 @@ func (m windowsManager) setPrimaryDNS(resolvers []netip.Addr, domains []dnsname.
 		domStrs = append(domStrs, dom.WithoutTrailingDot())
 	}
 
-	key4, err := m.openKey(m.ifPath(ipv4RegBase))
-	if err != nil {
+	if err := tsLUID.SetDNS(winipcfg.AddressFamily(windows.AF_INET), ips4, domStrs); err != nil {
 		return err
 	}
-	defer key4.Close()
 
-	if len(ipsv4) == 0 {
-		if err := delValue(key4, "NameServer"); err != nil {
-			return err
-		}
-	} else if err := key4.SetStringValue("NameServer", strings.Join(ipsv4, ",")); err != nil {
+	if err := tsLUID.SetDNS(winipcfg.AddressFamily(windows.AF_INET6), ips6, domStrs); err != nil {
 		return err
 	}
 
-	if len(domains) == 0 {
-		if err := delValue(key4, "SearchList"); err != nil {
-			return err
-		}
-	} else if err := key4.SetStringValue("SearchList", strings.Join(domStrs, ",")); err != nil {
+	// Disable LLMNR on the Tailscale interface. We don't do
+	// multicast, and we certainly don't do LLMNR, so it's pointless
+	// to make Windows try it.
+	//
+	// TODO(maisem): The same effect can be achieved by setting
+	// winipcfg.DnsInterfaceSettings.EnableLLMNR to 0. Currently upstream does
+	// not support setting the DnsInterfaceSettingsFlagEnableLLMNR flag, use it
```

Comment:
Is there an upstream bug/PR somewhere? Link?

Reply:
Not yet, will file.